xarray/core/dataarray.py

```diff
@@ -2413,6 +2413,64 @@ def differentiate(self, coord, edge_order=1, datetime_unit=None):
             coord, edge_order, datetime_unit)
         return self._from_temp_dataset(ds)
 
+    def cov(self, other, dim=None):
+        """Compute covariance between two DataArray objects along a shared dimension.
+
+        Parameters
+        ----------
+        other: DataArray
+            The other array with which the covariance will be computed
+        dim: The dimension along which the covariance will be computed
+
+        Returns
+        -------
+        covariance: DataArray
+        """
+        # 1. Broadcast the two arrays
+        self, other = broadcast(self, other)
+
+        # 2. Ignore the nans
+        valid_values = self.notnull() & other.notnull()
+        self = self.where(valid_values, drop=True)
```

Comment:
It would be best to avoid `drop=True` if possible. Dropping elements can really slow things down when using dask arrays, because determining the elements to drop requires computing the arrays. In contrast, if we avoid `drop=True` we can build a lazy computation graph.